src/format/parse.rs

```diff
@@ -316,8 +340,10 @@ where
                     Minute => (2, false, Parsed::set_minute),
                     Second => (2, false, Parsed::set_second),
                     Nanosecond => (9, false, Parsed::set_nanosecond),
-                    Timestamp => (usize::MAX, false, Parsed::set_timestamp),
-
+                    Timestamp => {
+                        let next_size = get_fixed_item_len(next_item).unwrap_or(0);
+                        (s.len() - next_size, false, Parsed::set_timestamp)
```

Comment:
Using `s.len()` here is problematic for parsing any formats where the timestamp does not terminate at the end of the string. 

For example, I have seen a format that is like `%s%3f%z` - so `123456789+0000` - in this case you hit `%s` consuming all of `123456789` because it is limited to the max of `123456789+0`.

Another case I've seen is `Date(%s%3f)` - so `Date(123456789)` - in this case you hit `%s` consuming all of `1234567` because that is the max, but `%3f` is left with three extra characters.

It seems the only real way would be to do something like `scan::number` to read however many chars `Timestamp` would capture and then go from there. Sadly, the return of `scan::number` doesn't currently include the number of characters read, instead returning a new slice pointer so that is not a trivial fix.

I don't know if it's really necessary to fix this for those formats, if they should just be a separate issue, or if they should just not be fixed by the library and left to consumers to implement tricky logic.